rt_utils/rtstruct.py

```diff
@@ -55,7 +57,7 @@ def add_roi(
             )
 
         self.ds.ROIContourSequence.append(ds_helper.create_roi_contour(roi_data, self.series_data))
-        self.ds.StructureSetROISequence.append(ds_helper.create_structure_set_roi(roi_data))
+        self.ds.StructureSetROISequence.append(ds_helper.create_structure_set_roi(roi_data, ROIGenerationAlgorithm))
```

Comment:
Lets package this in with the roi_data data class like the other parameters. This way we can access it all from the roi_data variable and won't need to add extra parameters to multiple functions